cmd/node/factory/structs.go

```diff
@@ -1878,6 +1896,7 @@ func newMetaBlockProcessor(
 		RequestHandler:               requestHandler,
 		Core:                         coreServiceContainer,
 		ValidatorStatisticsProcessor: validatorStatisticsProcessor,
+		Rounder:                      rounder,
```

Comment:
Actually there are many methods which needs current round index. We can reduce the number of parameters for all these method by injecting rounder or at least an interface with one method Index()

shardBlock

CreateBlockBody
createMiniBlocks
createAndProcessCrossMiniBlocksDstMe
getOrderedMetaBlocks
CreateMbsAndProcessCrossShardTransactionsDstMe
CreateMbsAndProcessTransactionsFromMe
CreateBlockHeader
checkAndRequestIfMetaHeadersMissing

metaBlock

checkAndRequestIfShardHeadersMissing
getOrderedHdrs
requestHeadersIfMissing
CreateBlockBody
createShardInfo
checkAndProcessShardMiniBlockHeader
CreateBlockHeader

Reply:
I have added it because I needed for isShardStuck method. But I will extract this method into another component together with the map lastHdrs, and I will remove the rounder from processor.